### PR TITLE
sessionctx, executor: add max_allowed_packet verification

### PR DESCRIPTION
**Pull request by iamzhoug37**
### What problem does this PR solve? 
add verification for system variable max_allowed_packet

### CODE REVIEW

File: sessionctx/variable/varsutil.go [hunk 1]
```diff
@@ -244,6 +244,8 @@ func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string,
 		return checkUInt64SystemVar(name, value, 0, 2, vars)
 	case InnodbLockWaitTimeout:
 		return checkUInt64SystemVar(name, value, 1, 1073741824, vars)
+	case MaxAllowedPacket:
+		return checkUInt64SystemVar(name , value , 1024, 1073741824 , vars)
```
[the remark this answers is not in the document]
Comment (zimulala) [reply sessionctx/variable/varsutil.go]: We can add the comment like "See https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_allowed_packet".

[the remark this answers is not in the document]
Comment (iamzhoug37) [reply sessionctx/variable/varsutil.go]: ok,thank you.

**Comment by zz-jason** [comment]
@iamzhoug37 Please cherrypick this patch to the release-2.1 branch

**Pull request merged**